File: 1814-count-nice-pairs-in-an-array/1814-count-nice-pairs-in-an-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var countNicePairs = function(nums) {
  const MOD = 1e9 + 7;
  const arr = nums.map(num => num - Number(String(num).split("").reverse().join("")));
  const map = new Map();
  let count = 0;
  
  for (let i = 0; i < arr.length; i++) {
    const cur = arr[i];

    if (map.has(cur)) {
      count = (count + map.get(cur)) % MOD;
    }
    
    map.set(cur, (map.get(cur) || 0) + 1);
  }
  
  return count;
};

File: 1814-count-nice-pairs-in-an-array/1814-count-nice-pairs-in-an-array.py
class Solution:
  def countNicePairs(self, nums: List[int]) -> int:
    MOD = 10 ** 9 + 7
    arr = [num - int(str(num)[::-1]) for num in nums]
    count = 0
    freq = {}
    
    for cur in arr:
      if cur in freq:
        count = (count + freq[cur]) % MOD
      
      freq[cur] = freq.get(cur, 0) + 1
      
    return count
